sunbeam-python/sunbeam/plugins/validation/plugin.py

```diff
@@ -78,12 +80,57 @@ def validate_schedule(schedule: str) -> tuple[bool, Optional[str]]:
     t1 = cron.get_next()
     t2 = cron.get_next()
     if t2 - t1 <= MINIMAL_PERIOD:
-        return (
-            False,
-            "Cannot schedule periodic check to run faster than every 15 minutes.",
+        raise click.ClickException(
+            "Cannot schedule periodic check to run faster than every 15 minutes."
         )
 
-    return True, f"Setting schedule to {schedule}"
+    return schedule
+
+
+@dataclass()
+class Config:
+    """Represents config updates provided by the user.
+
+    None values mean the user did not provide them.
+    """
+
+    schedule: Optional[str] = None
+
+
+def parse_config_args(args: List[str]) -> Dict[str, str]:
+    """Parse key=value args into a valid dictionary of key: values.
+
+    Raise a click bad argument error if errors (only checks syntax here).
+    """
+    config = {}
+    for arg in args:
+        split_arg = arg.split("=", 1)
+        if len(split_arg) == 1:
+            raise click.ClickException("syntax: key=value")
+        key, value = split_arg
+        if key in config:
+            raise click.ClickException(
+                f"{key} parameter seen multiple times.  Only provide it once."
+            )
+        config[key] = value
```

Comment:
Now I see we are using key-value pairs as inputs instead of arguments. My suggestion below would save us from implementing this logic and delegate the responsibility to click and argparse.